accessories/fan.js

```diff
@@ -30,15 +30,14 @@ function HomeAssistantFan(log, data, client) {
   } else {
     this.serial = data.entity_id;
   }
-  this.client = client;
-  this.log = log;
-
-  var speedList = data.attributes.speed_list;
-  if (speedList) {
+  if (data.attributes.speed_list) {
```

Comment:
I apologize for my lack of experience with JS, but is it not necessary to check that `data.attributes` is also defined?

Reply:
I see from my stack trace in #283 that the culprit is actually `speed_list`, so I'm going to assume that the `attributes` check isn't necessary. If that's the case, I'm not sure why the fields above are checking for `attributes`. Would you mind clarifying how this works? Thanks!

Reply:
Your original trace in #283 showed that `speed_list` didn't exist when it called `getRotationSpeed`  at line 99.

Checking for 'data.attributes.speed_list' accomplishes the same as checking for 'data', then checking for 'attributes', then checking for 'speed_list'; just less code.